### PR TITLE
.ci/build.sh: Make .ci/build.sh re-runnable

### PR DESCRIPTION
**Pull request by Prince-sharma**
This remove private,_site,public
after partial or complete build.

Closes https://github.com/coala/community/issues/184

### CODE REVIEW

**Comment by li-boxuan** [comment]
You mentioned on gitter that you tried to add `rm -rf private _site public` and `rm db.sqlite3` on `.ci/build.sh`, which doesn't work. Can you elaborate on it? Where did you put and what was the error?

**Comment by Prince-sharma** [comment]
@li-boxuan if I simply clone the repo and add `rm -rf private _site public`, `rm db.sqlite3` at the end of `.ci/build.sh`.
and then run `pytest` then  I get the error :

**Comment by Prince-sharma** [comment]
```python
Traceback (most recent call last):
  File "/home/prince/anaconda3/lib/python3.6/logging/config.py", line 558, in configure
    handler = self.configure_handler(handlers[name])
  File "/home/prince/anaconda3/lib/python3.6/logging/config.py", line 731, in configure_handler
    result = factory(**kwargs)
  File "/home/prince/anaconda3/lib/python3.6/logging/__init__.py", line 1030, in __init__
    StreamHandler.__init__(self, self._open())
  File "/home/prince/anaconda3/lib/python3.6/logging/__init__.py", line 1059, in _open
    return open(self.baseFilename, self.mode, encoding=self.encoding)
FileNotFoundError: [Errno 2] No such file or directory: '/home/prince/code/coala/community/_site/community.log'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/bin/pytest", line 11, in <module>
    sys.exit(main())
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/_pytest/config/__init__.py", line 55, in main
    config = _prepareconfig(args, plugins)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/_pytest/config/__init__.py", line 180, in _prepareconfig
    pluginmanager=pluginmanager, args=args
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/hooks.py", line 258, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/callers.py", line 196, in _multicall
    gen.send(outcome)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/_pytest/helpconfig.py", line 89, in pytest_cmdline_parse
    config = outcome.get_result()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/callers.py", line 76, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/callers.py", line 180, in _multicall
    res = hook_impl.function(*args)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/_pytest/config/__init__.py", line 612, in pytest_cmdline_parse
    self.parse(args)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/_pytest/config/__init__.py", line 777, in parse
    self._preparse(args, addopts=addopts)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/_pytest/config/__init__.py", line 739, in _preparse
    early_config=self, args=args, parser=self._parser
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/hooks.py", line 258, in __call__
    return self._hookexec(self, self._nonwrappers + self._wrappers, kwargs)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/manager.py", line 67, in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/manager.py", line 61, in <lambda>
    firstresult=hook.spec_opts.get('firstresult'),
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/callers.py", line 201, in _multicall
    return outcome.get_result()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/callers.py", line 76, in get_result
    raise ex[1].with_traceback(ex[2])
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pluggy/callers.py", line 180, in _multicall
    res = hook_impl.function(*args)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pytest_django/plugin.py", line 252, in pytest_load_initial_conftests
    _setup_django()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/pytest_django/plugin.py", line 151, in _setup_django
    django.setup()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/django/__init__.py", line 22, in setup
    configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/django/utils/log.py", line 75, in configure_logging
    logging_config_func(logging_settings)
  File "/home/prince/anaconda3/lib/python3.6/logging/config.py", line 795, in dictConfig
    dictConfigClass(config).configure()
  File "/home/prince/anaconda3/lib/python3.6/logging/config.py", line 566, in configure
    '%r: %s' % (name, e))
ValueError: Unable to configure handler 'communityHandler': [Errno 2] No such file or directory: '/home/prince/code/coala/community/_site/community.log'
```

**Comment by Prince-sharma** [comment]
since there is no dir `_site` which is stated in the above error.
but if I execute `bash .ci/build.sh`(which creates the dir `_site`,`private`,`public`) and then execute `pytest` it works fine and gives coverage of 94.6%

**Comment by Prince-sharma** [comment]
also while running `.ci/build.sh` I get the following error:

```python
+ mkdir private _site public
+ META_REVIEW_DATA=meta_review.json
+ EXPORTED_DATA='static/tasks.yaml static/instances.yaml static/meta_review.json'
+ ISSUES_JSON=issues.json
+ python manage.py fetch_deployed_data --allow-failure _site static/tasks.yaml static/instances.yaml static/meta_review.json
+ [[ -n '' ]]
+ python manage.py fetch_deployed_data _site issues.json --repo-name gh-board
Traceback (most recent call last):
  File "/home/prince/code/coala/community/data/management/commands/fetch_deployed_data.py", line 40, in handle
    r.raise_for_status()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/requests/models.py", line 939, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: https://prince-sharma.github.io/gh-board/issues.json

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "manage.py", line 24, in <module>
    execute_from_command_line(sys.argv)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/django/core/management/__init__.py", line 364, in execute_from_command_line
    utility.execute()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/django/core/management/__init__.py", line 356, in execute
    self.fetch_command(subcommand).run_from_argv(self.argv)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/django/core/management/base.py", line 283, in run_from_argv
    self.execute(*args, **cmd_options)
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/django/core/management/base.py", line 330, in execute
    output = self.handle(*args, **options)
  File "/home/prince/code/coala/community/data/management/commands/fetch_deployed_data.py", line 45, in handle
    r.raise_for_status()
  File "/home/prince/.local/share/virtualenvs/coala-qMtnnkwi/lib/python3.6/site-packages/requests/models.py", line 939, in raise_for_status
    raise HTTPError(http_error_msg, response=self)
requests.exceptions.HTTPError: 404 Client Error: Not Found for url: http://community.coala.io/issues.json
```

**Comment by li-boxuan** [comment]
> if I simply clone the repo and add rm -rf private _site public, rm db.sqlite3 at the end of .ci/build.sh. and then run pytest then I get the error

Why not put them at the beginning of .ci/build.sh?

**Comment by Prince-sharma** [comment]
I tried putting the `rm` commands at the beginning of `.ci/build.sh` but it gives the same error as in putting them at the end.

**Comment by Prince-sharma** [comment]
Also when I did a fresh start i.e cloned the repo https://github.com/coala/community, installed coala-bears using `pipenv install coala-bears`, installed dependencies using `pip3 install -r test-requirements.txt -r requirements.txt`  and then executed `pytest`(no changes done in the code) then too it gave the same error as mentioned above.

**Comment by li-boxuan** [comment]
The error you are facing here: https://github.com/coala/community/pull/186#issuecomment-416067492 is due to the issue https://github.com/coala/community/issues/185
If you don't want to fork and configure gh-board, you might want to solve that issue first.

**Comment by Prince-sharma** [comment]
@li-boxuan I got 100% branch coverage when ran locally,then why is CI giving error related to `pytest`.

following is the output of `pytest`



```
================================================================ test session starts =================================================================
platform linux -- Python 3.6.5, pytest-3.6.4, py-1.5.4, pluggy-0.7.1
Django settings: community.settings (from ini file)
rootdir: /home/prince/code/coala/community, inifile: setup.cfg
plugins: requests-mock-1.5.2, xdist-1.22.5, travis-fold-1.3.0, timeout-1.3.1, reqs-0.0.7, reorder-0.1.1, profiling-1.3.0, mock-1.10.0, instafail-0.3.0, forked-0.2, error-for-skips-1.0.0, env-0.6.2, django-3.3.3, cov-2.5.1
timeout: 35.0s
timeout method: signal
timeout func_only: False
collected 58 items                                                                                                                                   

requirements.txt .                                                                                                                             [  1%]
test-requirements.txt .                                                                                                                        [  3%]
data/tests/test_contrib_data.py .                                                                                                              [  5%]
data/tests/test_issues.py .                                                                                                                    [  6%]
data/tests/test_management_commands.py ...                                                                                                     [ 11%]
data/tests/test_merge_requests.py .                                                                                                            [ 13%]
data/tests/test_models.py ...............                                                                                                      [ 38%]
data/tests/test_views.py ...                                                                                                                   [ 43%]
gamification/tests/test_activity.py ...                                                                                                        [ 48%]
gamification/tests/test_activity_points.py ...                                                                                                 [ 53%]
gamification/tests/test_labels.py .                                                                                                            [ 55%]
gamification/tests/test_management_commands.py ...                                                                                             [ 60%]
gamification/tests/test_models.py ...................                                                                                          [ 91%]
gamification/tests/test_points.py .                                                                                                            [ 93%]
gamification/tests/test_views.py ...                                                                                                           [ 98%]
meta_review/handler.py .                                                                                                                       [100%]
Profiling (from /home/prince/code/coala/community/prof/combined.prof):
Mon Aug 27 13:31:38 2018    /home/prince/code/coala/community/prof/combined.prof

         536948 function calls (524235 primitive calls) in 2.736 seconds

   Ordered by: cumulative time
   List reduced from 1728 to 20 due to restriction <20>

   ncalls  tottime  percall  cumtime  percall filename:lineno(function)
       60    0.000    0.000    2.720    0.045 runner.py:105(pytest_runtest_call)
    52/48    0.001    0.000    1.539    0.032 <frozen importlib._bootstrap>:966(_find_and_load)
     10/6    0.000    0.000    1.537    0.256 <frozen importlib._bootstrap>:936(_find_and_load_unlocked)
     12/9    0.000    0.000    1.531    0.170 <frozen importlib._bootstrap>:211(_call_with_frames_removed)
        6    0.000    0.000    1.530    0.255 <frozen importlib._bootstrap>:651(_load_unlocked)
        2    0.000    0.000    1.203    0.601 pytest_reqs.py:136(runtest)
        2    0.000    0.000    1.195    0.597 pytest_reqs.py:119(get_requirements)
        5    0.000    0.000    1.121    0.224 <frozen importlib._bootstrap_external>:672(exec_module)
       57    0.000    0.000    1.101    0.019 unittest.py:182(runtest)
       57    0.001    0.000    1.100    0.019 testcases.py:195(__call__)
      984    0.001    0.000    1.075    0.001 <frozen importlib._bootstrap>:997(_handle_fromlist)
        2    0.000    0.000    1.075    0.537 download.py:321(__init__)
        2    0.001    0.000    1.072    0.536 download.py:63(user_agent)
       57    0.000    0.000    1.071    0.019 case.py:652(__call__)
        2    0.000    0.000    1.071    0.535 {built-in method builtins.__import__}
       57    0.004    0.000    1.071    0.019 case.py:570(run)
       12    0.001    0.000    1.061    0.088 {built-in method builtins.exec}
        1    0.000    0.000    1.059    1.059 distro.py:29(<module>)
        1    0.000    0.000    1.057    1.057 distro.py:530(__init__)
        1    0.000    0.000    1.054    1.054 distro.py:908(_get_lsb_release_info)



----------- coverage: platform linux, python 3.6.5-final-0 -----------
Name                                      Stmts   Miss Branch BrPart  Cover   Missing
-------------------------------------------------------------------------------------
community/__init__.py                         0      0      0      0   100%
community/config.py                           3      0      0      0   100%
community/filters.py                          4      0      0      0   100%
community/settings.py                        23      0      0      0   100%
community/wsgi.py                             4      0      0      0   100%
data/__init__.py                              0      0      0      0   100%
data/apps.py                                  3      0      0      0   100%
data/contrib_data.py                         32      0      4      0   100%
data/issues.py                               50      0      8      0   100%
data/merge_requests.py                       57      0     10      0   100%
data/models.py                               56      0      2      0   100%
data/newcomers.py                            11      0      2      0   100%
data/urls.py                                  3      0      0      0   100%
data/webservices.py                           5      0      0      0   100%
gamification/__init__.py                      0      0      0      0   100%
gamification/data/config.py                  51      0      0      0   100%
gamification/data/points.py                   3      0      0      0   100%
gamification/labels.py                       17      0     19      0   100%
gamification/models.py                       73      0     21      0   100%
gamification/participants.py                  7      0      2      0   100%
gamification/points.py                       12      0      6      0   100%
gamification/process/__init__.py              0      0      0      0   100%
gamification/process/activity.py              9      0      4      0   100%
gamification/process/activity_points.py       8      0      0      0   100%
gamification/process/update.py               55      0     13      0   100%
meta_review/__init__.py                       0      0      0      0   100%
meta_review/apps.py                           3      0      0      0   100%
meta_review/models.py                        38      0      0      0   100%
meta_review/urls.py                           3      0      0      0   100%
-------------------------------------------------------------------------------------
TOTAL                                       530      0     91      0   100%

Required test coverage of 100% reached. Total coverage: 100.00%

============================================================= 60 passed in 71.59 seconds =============================================================

```

**Comment by jayvdb** [comment]
@Prince-sharma , please put Mohan sync changes in a separate commit, as they are not part of your changes

**Comment by li-boxuan** [comment]
Please also update your PR title.

File: .ci/build.sh [hunk 1]
```diff
@@ -48,3 +50,5 @@ python manage.py dumpdata meta_review > _site/$META_REVIEW_DATA
 
 python manage.py collectstatic --noinput
 python manage.py distill-local public --force
+
+rm db.sqlite3
```
Comment (jayvdb) [inline .ci/build.sh]: add this to your `rm -rf` above.

this isnt isnt about removing artifacts after the build.

we want to keep these artifacts, so someone can debug the database if the built site looks strange.

but if they re-run the build, the database needs to be empty again.

**Comment by jayvdb** [comment]
ack https://github.com/coala/community/pull/186/commits/9bb5118bba0d45d865663a05c6dc0cbc539174c5

**Comment by jayvdb** [comment]
ack https://github.com/coala/community/pull/186/commits/5b5a6d71c5928829a394175c813f54c922475f90

**Comment by jayvdb** [comment]
@gitmate-bot ff

**Pull request merged**